pyuvdata/calfits.py

```diff
@@ -273,8 +285,13 @@ def write_calfits(self, filename, spoof_nonessential=False,
             hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
 
         else:
-            prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
-            hdulist = fits.HDUList([prihdu, ant_hdu])
+            if self.total_quality_array is not None:
+                prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+                sechdu = fits.ImageHDU(data=secdata, header=sechdr)
+                hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
+            else:
+                prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+                hdulist = fits.HDUList([prihdu, ant_hdu])
 
```

Comment:
Is it possible to put it in the secondary HDU when we have a delay type (i.e. do the array shapes work for that)? If so, I'd put it there. Otherwise having it in it's own HDU is probably the only option.